pallets/parachain-staking/src/lib.rs

```diff
@@ -2042,53 +2058,25 @@ pub mod pallet {
 			log::trace!("{} Candidates for {} Collator seats", candidates.len(), top_n);
 
 			// Order candidates by their total stake (greatest to least)
-			candidates.sort_by(|a, b| b.amount.cmp(&a.amount));
+			//
+			// NOTE: Resorting is required if a collator's total stake just changed before
+			// updating the selected candidates.
+			candidates.sort_by(|a, b| b.cmp(&a));
```

Comment:
Suggesting changes from the only clippy warning:

```suggestion
			candidates.sort_by(|a, b| b.cmp(a));
```